File: Data Structures & Algorithms/sliding-window-maximum/submission-0.cpp
class Solution {
public:
    std::vector<int> maxSlidingWindow(std::vector<int>& nums, int k) {
        int max=INT_MIN, l=0, r=k;
        std::vector<int> msw;
        for (int i=0;i<k;i++){
            max=std::max(max,nums[i]);
        }
        msw.push_back(max);
        while (r<nums.size()){
            if(nums[l]==max){
                max=INT_MIN;
                for (int i=l;i<l+k;i++){
                    max=std::max(max,nums[i+1]);
                }
            }
            max=std::max(max, nums[r]);
            l++, r++;
            msw.push_back(max);
        }
        return msw;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
  def maxProfit(self, prices: List[int]) -> int:
    if len(prices) == 1:
      return 0
    res = 0
    l = 0
    r = 1
    while len(prices) > r:
      if prices[r]-prices[l] > 0:
        res = max(res, prices[r]-prices[l])
      elif prices[r]-prices[l] < 0:
        l = r
      r+=1
    return res
